src/main/java/io/bittiger/ads/activity/AdsSelect.java

```diff
@@ -0,0 +1,65 @@
+package io.bittiger.ads.activity;
+
+import io.bittiger.ads.model.Ad;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
+import java.lang.String;
+
+public class AdsSelect {
+    private static AdsSelect instance = null;
+    
+    protected AdsSelect() {
+    }
+
+    public static AdsSelect getInstance() {
+        if (instance == null) {
+            instance = new AdsSelect();
+        }
+        return instance;
+    }
+
+    public List<Ad> getMatchedAds(String[] keywords) {
+        List<Ad> ads = new ArrayList<Ad>();
+        if (keywords == null || keywords.length == 0) {
+            return ads;
+        }
+        
+        try {
+            MemcachedClient cache = AdsDao.getCache();
+            Set<ad> set = new HashSet<>();
+            for (String keyword : keywords) {
+                Set<ad> curr = (Set<ad>) cache.getAds("inv" + keyword);
+                if (curr == null) {
+                    curr = traverseFwdIndex(keyword);
+                }
+                
+                set.addAll(curr);
+            }
+            ads.addAll(set);
+            return ads;

```

Comment:
When all the candidate ads are selected, we should calculate the relevant score before they're returned.